lib/defaults/labels.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaults
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	// KubernetesRoleLabel is the Kubernetes node label with system role
+	KubernetesRoleLabel = "gravitational.io/k8s-role"
+
+	// KubernetesAdvertiseIPLabel is the kubernetes node label of the advertise IP address
+	KubernetesAdvertiseIPLabel = "gravitational.io/advertise-ip"
+
+	// RunLevelLabel is the Kubernetes node taint label representing a run-level
+	RunLevelLabel = "gravitational.io/runlevel"
+
+	// KubernetesReconcileLabel is the kubernetes node label to control the reconcile process for the node
+	KubernetesReconcileLabel = "gravitational.io/reconcile"
```

Comment:
I have a small concern with a too broad of a name for this label - it should hint at labels as both it and the values are node label-specific - e.g. `reconcile-labels` or even flipping it and embedding the context in the label name: `node-labels.gravitational.io/mode` and naming the modes `reconcile`, `ensureexists` and empty or `none` (for no reconciliation)?

Or did you want to reserve the name for a more general application that might potentially extend to future types of reconciliation loops?

Reply:
I thought about more general behavior that might potentially extend to future types of reconciliation loops. And I thought about the name `reconcile.gravitational.io/mode`, maybe it would be better. What do you think about it?

Reply:
I think it could be confusing if this had an additional application (outside of labels) as none of the current values are label-specific. So you would either have to embed the label detail inside of the mode values - e.g. `EnsureLabelExists` or `ReconcileLabels` or go with a more specific label name: `node-label.gravitational.io/mode` similar to [addonmanager](https://github.com/kubernetes/kubernetes/tree/4f254674fa7f3752fcf25cbd17ad6dc2485c124c/cluster/addons/addon-manager)

Reply:
I suggest not to use the word "label" in the name and use a more generic name, because later we can add an additional type of reconciliation that will rely on this label.